src/generator/AutoRest.Go/Model/MethodGo.cs

```diff
@@ -41,6 +43,18 @@ internal void Transform(CodeModelGo cmg)
       PackageName = cmg.Namespace;
       NextAlreadyDefined = NextMethodExists(cmg.Methods.Cast<MethodGo>());
 
+      var apiVersionParam =
+        from p in Parameters
+        let name = p.SerializedName.Value
+        where name != null && name.IsApiVersion()
+        select p.DefaultValue.Value?.Trim(new[]{'"'});
+
+      APIVersion = apiVersionParam.SingleOrDefault();
+      if(APIVersion == default(string))
```

Comment:
Interesting.  I'm fine with either approach I would just add a comment with this explanation.